contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -0,0 +1,933 @@
+                        tickUpper: upperTick,
+                        amount0Desired: _wethBalance,
+                        amount1Desired: _oethbRequired,
+                        amount0Min: 0,
+                        amount1Min: 0,
+                        recipient: address(this),
+                        deadline: block.timestamp,
+                        sqrtPriceX96: 0
+                    })
+                );
+
+            _wethAmountSupplied = __wethAmountSupplied;
+            _oethbAmountSupplied = __oethbAmountSupplied;
+            tokenId = _mintedTokenId;
+        } else {
+            (
+                ,
+                uint256 __wethAmountSupplied,
+                uint256 __oethbAmountSupplied
+            ) = positionManager.increaseLiquidity(
+                    /** amount0Min & amount1Min are left at 0 because slippage protection is ensured by the
+                     * _checkForExpectedPoolPrice
+                     */
+                    INonfungiblePositionManager.IncreaseLiquidityParams({
+                        tokenId: tokenId,
+                        amount0Desired: _wethBalance,
+                        amount1Desired: _oethbRequired,
+                        amount0Min: 0,
+                        amount1Min: 0,
+                        deadline: block.timestamp
+                    })
+                );
+
+            _wethAmountSupplied = __wethAmountSupplied;
+            _oethbAmountSupplied = __oethbAmountSupplied;
+        }
+
+        _updateUnderlyingAssets();
+        emit LiquidityAdded(
+            _wethBalance, // wethAmountDesired
+            _oethbRequired, // oethbAmountDesired
+            _wethAmountSupplied, // wethAmountSupplied
+            _oethbAmountSupplied, // oethbAmountSupplied
+            tokenId, // tokenId
+            underlyingAssets
+        );
+
+        // burn remaining OETHb
+        _burnOethbOnTheContract();
+        positionManager.approve(address(clGauge), tokenId);
+        clGauge.deposit(tokenId);
+    }
+
+    // slither-disable-end reentrancy-no-eth
+
+    /**
+     * @dev Check that the Aerodrome pool price is within the expected
+     *      parameters.
+     *      This function ignores whether the strategy contract has liquidity
+     *      position in the pool.
+     */
+    function _checkForExpectedPoolPrice() internal {
+        uint160 _currentPrice = getPoolX96Price();
+        // check we are in inspected tick range
+
+        if (_currentPrice < sqrtRatioX96Tick0 ||
+            _currentPrice > sqrtRatioX96Tick1) {
+
+            int24 currentTick = getCurrentTradingTick();
+            revert OutsideExpectedTickRange(
+                currentTick,
+                lowerTick,
+                upperTick
+            );
+        }
+
+        // If token addresses were reversed estimateAmount0 would
+        // be required here
+        uint256 _normalizedWethAmount = 1 ether;
+        uint256 _correspondingOETHAmount = helper.estimateAmount1(
```

Comment:
🟠🟣 This method of calculating percentages can run into infinities and then revert.

Essentially we are saying "given one of something on one side, how much do we have on the other side", and this goes to infinity as the WETH amount gets to zero. Fortunately the helper starts reverting somewhere under returning a 2^128 number, and that revert makes this revert. But it's for completeness it's worth noting that we could get a third kind of revert out of this function than we were expecting.

Reply:
I've added a comment just in case we are tracing a transaction error in the future, so it helps is debug it:
https://github.com/OriginProtocol/origin-dollar/pull/2184/commits/8e22f76d984d50c7cfd6c0ee04dd42b38a98f4e9